src/c/device-coap.c

```diff
@@ -99,50 +103,130 @@ static bool coap_init
   return true;
 }
 
-static bool coap_get_handler
-(
-  void *impl,
-  const devsdk_device_t *device,
-  uint32_t nreadings,
-  const devsdk_commandrequest *requests,
-  devsdk_commandresult *readings,
-  const iot_data_t *options,
-  iot_data_t **exception
-)
-{
-  (void) impl;
-  (void) device;
-  (void) nreadings;
-  (void) requests;
-  (void) readings;
-  (void) options;
-
-  *exception = iot_data_alloc_string (NOT_SUPPORTED_TEXT, IOT_DATA_REF);
-  return false;
+static bool coap_get_handler(void *impl, const devsdk_device_t *device,
+                             uint32_t nreadings,
+                             const devsdk_commandrequest *requests,
+                             devsdk_commandresult *readings,
+                             const iot_data_t *options,
+                             iot_data_t **exception) {
+  bool successful_get_request = true;
+  coap_driver *driver = (coap_driver *)impl;
+  int ret = EXIT_FAILURE;
+  uint32_t i = 0;
+
+  if (device == NULL) {
+    iot_log_error(driver->lc, "COAP:Device is empty");
+    return successful_get_request;
+  }
+  successful_get_request = GetEndDeviceConfig(device->name);
+  if (successful_get_request != true) {
+    iot_log_error(driver->lc, "COAP:GetEndDeviceConfig failed");
+    return successful_get_request;
+  }
+
+  iot_log_debug(driver->lc, "COAP:Triggering auto events nreadings=%d\n",
```

Comment:
Note that this method is not only called for auto events but also for non-automatic reading requests

Reply:
Updated as suggested.